dynamic/discovery/discovery.go

```diff
@@ -89,6 +89,23 @@ func (rm *ResourceMap) GetKind(apiVersion, kind string) (result *APIResource) {
 	return gv.kinds[kind]
 }
 
+func (rm *ResourceMap) HasSubresource(parentResource *APIResource, subresourceKey string) bool {
```

Comment:
As mentioned above, I would prefer that HasSubresource is a method on `APIResource`, not on `ResourceMap`. That way, APIResource includes everything you need to know, and you don't have to go back to the map to ask more questions if you already have the APIResource (which is already available inside every ResourceClient).